src/crafting.cpp

```diff
@@ -52,8 +52,42 @@ static bool crafting_allowed( const player &p, const recipe &rec )
     }
 
     if( p.lighting_craft_speed_multiplier( rec ) <= 0.0f ) {
-        add_msg( m_info, _( "You can't see to craft!" ) );
-        return false;
+        bool light_check = false;
+        bool dome_light = false;
+        if( g->m.veh_at( p.pos() ).has_value() &&
+            query_yn( "It's too dark to craft.  Would turning on a vehicle light on help?" ) ) {
+            vehicle *veh = &( g->m.veh_at( p.pos() )->vehicle() );
+            if( veh->has_part( p.pos(), "CONTROLS", false ) ||
```

Comment:
This whole bit about figuring out if the vehicle has lights and trying to turn them on seems like it would be a great method to add to the vehicle code.  Then this would just be
```
light_check = veh->try_to_turn_on_lights()
```

or something to that effect

Reply:
It would also be reuseable above in `activity_handlers::craft_do_turn`